### PR TITLE
feat(pipeline): support addons resources for pipeline

### LINKED ISSUE #650: Support addons with continuous delivery
### Existing system
Here is our existing CodePipeline layout.
pipeline

Within the build stage of the pipeline, we run `app package --output-dir infrastructure` to generate the CloudFormation template from the manifest file and addons template (see https://github.com/aws/amazon-ecs-cli-v2/issues/648)

Once the build stage completes, the `infrastructure/` directory is added as an artifact to the [PipelineBuiltArtifactBucket](https://github.com/aws/amazon-ecs-cli-v2/blob/master/templates/project/cf.yml#L76-L84).

### Required changes
1. We need to set the appropriate value for `AddonsTemplateURL` in the Test stage so that the template url for the nested stack can be constructed in the application stack. We're proposing to leverage [Fn::GetArtifactAtt](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/continuous-delivery-codepipeline-parameter-override-functions.html#w2ab1c13c20b7) with `ParameterOverrides` in each environment stage to retrieve the URL of the output artifact from the Build stage. So we would need to make the following change to [`pipeline_cfn.yml`](https://github.com/efekarakus/amazon-ecs-cli-v2/commit/b85c943421df2f4049d4c87610f2965408eef9e3):

```yaml
ParameterOverrides: |
  {
    "AddonsTemplateURL" : { "Fn::Join" : [ "/", [ "Fn::GetArtifactAtt" : ["BuildOutput", "URL"], "infrastructure", "{{$app}}.addons.stack.yml " ] ] }
  }
```

2. However, we can't set the `ParameterOverrides` if the application does not have any addons!
Therefore, we need to modify [stack/pipeline.go](https://github.com/aws/amazon-ecs-cli-v2/blob/master/internal/pkg/deploy/cloudformation/stack/pipeline.go) to pass information to denote that a particular application is expected to have addons.

### Parent story 
https://github.com/aws/amazon-ecs-cli-v2/issues/644

### PR DESCRIPTION
**Pull request by iamhopaul123**
fix #650

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/pipeline_init.go [hunk 1]
```diff
@@ -274,6 +302,35 @@ func (o *initPipelineOpts) createBuildspec() (string, error) {
 	return path, nil
 }
 
+func (o *initPipelineOpts) getArtifactBucket() ([]artifactBucket, error) {
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: nit:
```suggestion
func (o *initPipelineOpts) artifactBuckets() ([]artifactBucket, error) {
```

File: internal/pkg/cli/pipeline_init.go [hunk 2]
```diff
@@ -274,6 +302,35 @@ func (o *initPipelineOpts) createBuildspec() (string, error) {
...
+	proj, err := o.storeSvc.GetProject(o.ProjectName())
+	if err != nil {
+		return nil, fmt.Errorf("get project %s: %w", o.ProjectName(), err)
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: nit: I've been adding "metadata" whenever referring to errors from SSM, what do you think of:
```suggestion
		return nil, fmt.Errorf("get project metadata %s: %w", o.ProjectName(), err)
```

File: templates/cicd/buildspec.yml [hunk 3]
```diff
@@ -34,6 +34,17 @@ phases:
...
+      - |
+        for app in $apps; do
+          ADDONSFILE=./infrastructure/$app.addons.stack.yml
+          if [ -f "$ADDONSFILE" ]; then
+            tmp=$(mktemp)
+            timestamp=$(date +%s){{range $bucket := .ArtifactBuckets}}
+            aws s3 cp "$ADDONSFILE" "s3://{{$bucket.BucketName}}/manual/$timestamp/$app.addons.stack.yml";{{range $envName := $bucket.Environments}}
+            jq --arg a "https://{{$bucket.BucketName}}.s3-{{$bucket.Region}}.amazonaws.com/manual/$timestamp/$app.addons.stack.yml" '.Parameters.AddonsTemplateURL = $a' ./infrastructure/$app-{{$envName}}.params.json > "$tmp" && mv "$tmp" ./infrastructure/$app-{{$envName}}.params.json{{end}}{{end}}
+          fi
+        done;
```
Comment (efekarakus) [inline templates/cicd/buildspec.yml]: 👌 amazing! Thanks for diving deep on this 🌈

File: internal/pkg/cli/app_deploy.go [hunk 4]
```diff
@@ -446,7 +446,8 @@ func (o *appDeployOpts) showAppURI() error {
 // BuildAppDeployCmd builds the `app deploy` subcommand.
 func BuildAppDeployCmd() *cobra.Command {
 	vars := appDeployVars{
-		GlobalOpts: NewGlobalOpts(),
+		GlobalOpts:   NewGlobalOpts(),
+		enableAddons: true,
```
[the remark this answers is not in the document]
Comment (iamhopaul123) [reply internal/pkg/cli/app_deploy.go]: 😅

File: internal/pkg/cli/pipeline_init.go (hunk 2 above)
Comment (iamhopaul123) [reply internal/pkg/cli/pipeline_init.go]: I agree. And we should be consistent.

**Comment by iamhopaul123** [comment]
> Looks rad!
> 
> Our build execution role has permissions to put to s3?

Yes we have!

**Pull request merged**